### DISCUSSION TITLE
Docs tutorial screenshots

### DISCUSSION THREAD

**Discussion by khabdrick** [meshery/meshery#11750] [General]
I was working on a tutorial and I noticed that all the screenshots are placed in one folder irrespective of the tutorial. Is there a reason for this? 
It'll be tedious for someone trying to contribute to the tutorial to find the images and update if needed.
My suggestion is for every tutorial to have its own folder and all the screenshot will live there (if it's possible).
Another thing we could look at is using descriptive file names, which will help with SEO and collaboration too.
![2024-08-22_20-01](https://github.com/user-attachments/assets/a284c38c-4100-4a2a-8383-1c00f2101192)

What are your thoughts?

**Comment by Ganesh-403** [comment]
@khabdrick I think your suggestion makes sense from both a contributor and maintenance perspective.

Having a single shared screenshots folder can become difficult to navigate as the documentation grows—especially when multiple tutorials have similarly named images or need updates.

A page-bundle style structure such as:

```text
docs/
└── tutorials/
    └── install-meshery/
        ├── index.md
        ├── screenshot-1.webp
        ├── dashboard-overview.webp
        └── connection-settings.webp
```

has a few advantages:

- Images stay alongside the tutorial that uses them, making updates much easier.
- It reduces the chance of filename collisions.
- Contributors can immediately see which assets belong to a page.
- Descriptive filenames improve readability for contributors and are generally better than timestamp-based names.

Meshery's documentation already encourages using **page-bundle assets with relative paths** for documentation pages, while reserving `static/` for globally shared assets. That organization aligns well with keeping tutorial-specific screenshots next to the tutorial itself. :contentReference[oaicite:0]{index=0}

The only trade-off I can think of is that if the **same screenshot is intentionally reused across multiple tutorials**, a shared location avoids duplication. In that case, perhaps a hybrid approach would work well:

- **Tutorial-specific screenshots** → stored with the tutorial/page bundle.
- **Shared UI images, logos, and diagrams** → kept in a common assets directory.

That seems like a good balance between maintainability and avoiding duplicate assets.

Overall, I think this would make contributing to tutorials much easier. I'd be interested to hear whether there was an original reason for centralizing all screenshots, or whether it's simply something that has evolved over time.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
